vms/platformvm/vm_regression_test.go

```diff
@@ -80,17 +80,26 @@ func TestAddDelegatorTxOverDelegatedRegression(t *testing.T) {
 	)
 	require.NoError(err)
 
-	// trigger block creation
-	vm.ctx.Lock.Unlock()
-	require.NoError(vm.issueTxFromRPC(addValidatorTx))
-	vm.ctx.Lock.Lock()
+	blk0ID, err := vm.LastAccepted(t.Context())
+	require.NoError(err)
+	blk0, err := vm.GetBlock(t.Context(), blk0ID)
+	require.NoError(err)
 
-	// Accept addValidatorTx
-	require.NoError(buildAndAcceptStandardBlock(vm))
+	acceptedBlk1, err := block.NewBanffStandardBlock(
+		blk0.Timestamp().Add(time.Second),
+		blk0.ID(),
+		blk0.Height()+1,
+		[]*txs.Tx{addValidatorTx},
+	)
+	require.NoError(err)
+
+	blk1, err := vm.ParseBlock(t.Context(), acceptedBlk1.Bytes())
+	require.NoError(err)
+	require.NoError(blk1.Verify(t.Context()))
+	require.NoError(blk1.Accept(t.Context()))
 
 	// Advance the time
 	vm.clock.Set(validatorStartTime)
-	require.NoError(buildAndAcceptStandardBlock(vm))
```

Comment:
There seem to have been a couple of these blocks removed during the refactoring of this test. I think it's fine...

This test was added in: https://github.com/ava-labs/avalanchego/commit/38bd6eb0ee5c6e56425cdc3d1511f1b1b7fedcc4. It seems the the code now under test is: https://github.com/ava-labs/avalanchego/blob/24957e2afbdf28ec90fbe1a913eda3b314f443c6/vms/platformvm/txs/executor/staker_tx_verification_helpers.go#L200

If we remove that case, this test still fails. So I think it's fine.